pinery-lims/src/main/java/ca/on/oicr/pinery/lims/LimsAttribute.java

```diff
@@ -77,21 +78,21 @@ public String asString(boolean useEnumToString) {
     private static void validate() {
         Set<String> ins = new HashSet<>();
         Set<String> outs = new HashSet<>();
-        for (SampleAttribute sa : SampleAttribute.values()) {
-            if (Sets.intersection(ins, sa.in).isEmpty()) {
-                ins.addAll(sa.in);
+        for (LimsAttribute la : LimsAttribute.values()) {
+            if (Sets.intersection(ins, la.in).isEmpty()) {
+                ins.addAll(la.in);
             } else {
-                throw new RuntimeException("Duplicate: [" + sa.in.toString() + "]");
+                throw new RuntimeException("Duplicate: [" + la.in.toString() + "]");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Duplicate: [" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)